File: trunk/server/src/logic/prpc/CampType.go
package prpc

// enum CampType
const (
	CT_RED  = 0
	CT_BLUE = 1
	CT_MAX  = 2
)
const (
	K_CT_RED  = "CT_RED"
	K_CT_BLUE = "CT_BLUE"
	K_CT_MAX  = "CT_MAX"
)

func ToName_CampType(id int) string {
	switch id {
	case CT_RED:
		return "CT_RED"
	case CT_BLUE:
		return "CT_BLUE"
	case CT_MAX:
		return "CT_MAX"
	default:
		return ""
	}
}
func ToId_CampType(name string) int {
	switch name {
	case "CT_RED":
		return CT_RED
	case "CT_BLUE":
		return CT_BLUE
	case "CT_MAX":
		return CT_MAX
	default:
		return -1
	}
}

File: trunk/Client/Assets/Scripts/Common/Rpc/CampType.cs
public class CampType{
  static string[] _NAMES = new string[]{
    "CT_RED",
    "CT_BLUE",
    "CT_MAX"
  };
  public const int CT_RED = 0;
  public const int CT_BLUE = 1;
  public const int CT_MAX = 2;
  public static int ToId(string name){
    for(int i=0; i<NAMES.Length;++i){
      if(_NAMES[i] == name){
        return i;
      }
    }
    return -1;
  }
  public static string ToName(int id){
    if(id<0||id>=_NAMES.Length){
      return "";
    }
    return _NAMES[id];
  }
  public static string[] NAMES{ get {return _NAMES;} }
}
